### PR TITLE
refactor(parser): drop Buffer dependency

### PR DESCRIPTION
**Pull request by erfanium**

### CODE REVIEW

File: src/parser/utils.ts [hunk 1]
```diff
@@ -143,3 +143,97 @@ export function writeToBytes(
     bytes.length,
   );
 }
+
+export function utf8Slice(buf: Uint8Array, start: number, end: number) {
```
Comment (erfanium) [inline src/parser/utils.ts:147]: Can we replace this function with something native?

Comment (lucsoft) [reply src/parser/utils.ts:147]: Yes it was only some hacky stuff

Comment (erfanium) [reply src/parser/utils.ts:147]: So what is the alternative?

Comment (lucsoft) [reply src/parser/utils.ts:147]: I don't know I think I just yoinked it from the nodejs buffer Compact layer

**Pull request merged**